trackers_integration/tests/test_mantis.py

```diff
@@ -0,0 +1,193 @@
+            self.assertIn(expected_string, issue["description"])
+
+        # verify that LR has been added to TE
+        self.assertTrue(
+            LinkReference.objects.filter(
+                execution=self.execution_1,
+                url=result["response"],
+                is_defect=True,
+            ).exists()
+        )
+
+        # Close issue after test is finised.
+        close_issue(self.integration.rpc, new_issue_id)
+
+    def test_report_issue_from_test_execution_empty_baseurl_exception(self):
+        # simulate user clicking the 'Report bug' button in TE widget, TR page
+        bug_system = BugSystem.objects.create(  # nosec:B106:hardcoded_password_funcarg
+            name="Mantis for kiwitcms/test-mantis-integration Exception",
+            tracker_type="tcms.issuetracker.mantis.Mantis",
+            base_url="incorrect_url",
+            api_password=os.getenv("MANTIS_INTEGRATION_API_PASSWORD"),
+        )
+        integration = Mantis(bug_system, None)
+        result = self.rpc_client.Bug.report(
+            self.execution_1.pk, integration.bug_system.pk
+        )
+        self.assertIn("bug_report_page.php", result["response"])
+
+
+class TestMantisAPI(unittest.TestCase):
+    mantis_project = "Sample_Project"
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        client_secret = os.getenv("MANTIS_INTEGRATION_API_PASSWORD")
+        base_url = "http://localhost"
+        cls.api_instance = MantisAPI(base_url, client_secret)
+
+        test_issue_data = {
+            "summary": "Sample Issue",
+            "description": "Sample Issue Description",
+            "category": {"name": "General"},
+            "project": {"name": cls.mantis_project},
+        }
+
+        result = cls.api_instance.create_issue(test_issue_data)
+        cls.issue_id = result["issue"]["id"]
+
+    @classmethod
+    def tearDownClass(cls):
+        close_issue(cls.api_instance, cls.issue_id)
+        return super().tearDownClass()
+
+    def test_create_issue(self):
+        test_issue_data = {
+            "summary": "Sample Issue",
+            "description": "Sample Issue Description",
+            "category": {"name": "General"},
+            "project": {"name": self.mantis_project},
+        }
+
+        result = self.api_instance.create_issue(test_issue_data)
+        self.assertEqual(result["issue"]["summary"], "Sample Issue")
+
+        # Close work item after test is finished.
+        close_issue(self.api_instance, result["issue"]["id"])
+
+    def test_add_comment(self):
+        test_comment_body = {"text": "Test Comment"}
+        result = self.api_instance.add_comment(self.issue_id, test_comment_body)
+        self.assertEqual(result["note"]["text"], "Test Comment")
+        self.api_instance.delete_comment(self.issue_id, result["note"]["id"])
+
+
+def close_issue(instance, issue_id):
+    close_issue_body = {"status": {"name": "closed"}}
+    instance.update_issue(issue_id, close_issue_body)
```

Comment:
Note: unsure if this will work before we can test it.

Reply:
Works on local tests.

Reply:
Let's move this method into the `MantisAPI` class.